### PR TITLE
Fix file matching when dot appears before extension

### LINKED ISSUE #53: Confusing name for anonymous plugins
If you register an anonymous plugin inside fastify [this part](https://github.com/fastify/fastify-plugin/blob/master/index.js#L44) leads to confusing plugin names.

For example

```js
  fastify.register(fp((instance, opts, next) => {
    instance.decorate('anonymous', (a, b) => a + b)
    next()
  }))
```

will return `cemre` as the name of the plugin for me instead of the file name which was very confusing since it is my first name 😄

I think this is due to windows (the regex fails there) since all tests fail for me as well.

### PR DESCRIPTION
**Pull request by cemremengu**
Closes #53 

I didn't know how to add a unit test for this. Any ideas are appreciated.

There might be better ways to solve this as well.

### CODE REVIEW

File: test/stackParser.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+at loop (/home/leonardo/desktop/fastify-plugin/node_modules/function-loop/index.js:35:15)
+at TAP.runBeforeEach (/home/leonardo/desktop/fastify-plugin/node_modules/tap/lib/test.js:683:7)
+at TAP.processSubtest (/home/leonardo/desktop/fastify-plugin/node_modules/tap/lib/test.js:369:12)
+at TAP.process (/home/leonardo/desktop/fastify-plugin/node_modules/tap/lib/test.js:306:14)
+at TAP.sub (/home/leonardo/desktop/fastify-plugin/node_modules/tap/lib/test.js:185:10)
+at TAP.test (/home/leonardo/desktop/fastify-plugin/node_modules/tap/lib/test.js:209:17)`
+
+const anonymousStack = `Unable to parse this`
+
+test('should extract plugin name correctly from stack', t => {
```
Comment (mcollina) [inline test/stackParser.test.js]: nit: I don't think there is the need here to use `test`.

Comment (cemremengu) [reply test/stackParser.test.js]: Sorry, can you elaborate a bit? You mean it should be `t.test` ?

Comment (mcollina) [reply test/stackParser.test.js]: you can completely remove the `t.test` part, the script will work anyway.

Comment (cemremengu) [reply test/stackParser.test.js]: Thanks, I didn't know it could be used like that 👍

Comment (cemremengu) [reply test/stackParser.test.js]: @mcollina let me me know if this is ok to patch in when you have time

**Comment by cemremengu** [comment]
Lacking permissions

**Pull request merged**